lib/ui/common/SectionListView.dart

```diff
@@ -0,0 +1,71 @@
+
+
+import 'package:flutter/material.dart';
+
+abstract class ListViewSection {
+    IndexedWidgetBuilder itemBuilder();
+    int length();
+}
+
+class _SectionPosition {
+  final int offset;
+  final ListViewSection section;
+
+  _SectionPosition(this.offset, this.section);
+  int relativeOffset(int absoluteOffset) {
+      return absoluteOffset - offset;
+  }
+}
+
+class SectionedListView extends StatelessWidget implements ListViewSection {
+  final ScrollPhysics _physics;
+  final bool _shrinkWrap;
+  final List<_SectionPosition> _sectionPositions;
+  final int _allItemCount;
+
+  SectionedListView._(this._sectionPositions,this._allItemCount, this._shrinkWrap, this._physics);
+
+  factory SectionedListView({@required List<ListViewSection> sections, ScrollPhysics physics = const ScrollPhysics(), bool shrinkWrap = true}) {
+    int allItemCount = 0;
+    final sectionPositions = sections.map((section) {
+        int offset = allItemCount;
+        allItemCount += section.length();
+        return _SectionPosition(offset, section);
+    }).toList();
+    return  SectionedListView._(sectionPositions,allItemCount,shrinkWrap,physics);
+  }
+
+  _SectionPosition sectionForIndex(int index) {
+    for (var sectionPosition in _sectionPositions.reversed) {
```

Comment:
Same as the previous comment you can use `firstWhere` or `singleWhere`
```suggestion
    return _sectionPositions.reversed.firstWhere((
        sectionPosition) => sectionPosition.relativeOffset(index) >= 0,
        orElse: () => null);
```

Reply:
👍